time_capture/scripts/employee.py

```diff
@@ -0,0 +1,23 @@
+import frappe
+from frappe import _
+
+
+def before_validate(doc, method):
+	validate_expected_working_hours(doc)
+
+
+def validate_expected_working_hours(doc):
+	if not doc.date_of_joining == min(ewh.valid_from for ewh in doc.expected_working_hours):
```

Comment:
```suggestion
	if not doc.date_of_joining == min(getdate(ewh.valid_from) for ewh in doc.expected_working_hours):
```

Sometimes the `doc` my contain string dates and sometimes parsed `date` objects. `getdate` ensures that we always have a proper `date` object.

Import with: `from frappe.utils.data import getdate`